tensorflow/core/kernels/sparse_tensor_dense_matmul_op.cc

```diff
@@ -132,22 +147,39 @@ class SparseTensorDenseMatMulOp : public OpKernel {
       return;
     }
 
-#define MAYBE_ADJOINT(ADJ_A, ADJ_B)                                        \
-  if (adjoint_a_ == ADJ_A && adjoint_b_ == ADJ_B) {                        \
-    Status functor_status = functor::SparseTensorDenseMatMulFunctor<       \
-        Device, T, Tindices, ADJ_A,                                        \
-        ADJ_B>::Compute(ctx->eigen_device<Device>(), out->matrix<T>(),     \
-                        a_indices->matrix<Tindices>(), a_values->vec<T>(), \
-                        b->matrix<T>());                                   \
-    OP_REQUIRES_OK(ctx, functor_status);                                   \
+    switch (ndim) {
+#define MAYBE_ADJOINT(ADJ_A, ADJ_B, N)                               \
```

Comment:
@rmlarsen Passing b and output as `Tensor` instead of `TTypes<T,NDIM>::Tensor` will solve this. But I'm not sure whether this is proper, as when I read other codes, people tend to pass parameters as `TTypes<T,NDIM>::Tensor`. I'm not sure if there is some reason behind it that I am not aware of.